assets/app/features/upgrade/controllers/upgrade-database-configuration.controller.js

```diff
@@ -11,16 +11,92 @@
     angular.module('crowbarApp')
         .controller('UpgradeDatabaseConfigurationController', UpgradeDatabaseConfigurationController);
 
-    UpgradeDatabaseConfigurationController.$inject = [];
+    UpgradeDatabaseConfigurationController.$inject = ['upgradeFactory'];
     // @ngInject
-    function UpgradeDatabaseConfigurationController() {
+    function UpgradeDatabaseConfigurationController(upgradeFactory) {
         var vm = this;
         vm.databaseForm = {
             username: '',
             password: '',
             server: '',
-            port: 5432,
-            tablePrefix: ''
+            port: 5432
+        };
+
+        vm.running = false;
+        vm.spinnerVisible = false;
+        vm.completed = false;
+        vm.errors = [];
+
+        vm.createServer = createServer;
+        vm.connectServer = connectServer;
+
+        // functions
+
+        function createServer() {
+            var emptyFields = _.some(
```

Comment:
oh wait, ng-disabled has nothing to do with it, this was to protect against empty fields when clicking the button and indeed, that should be dealt separately I guess? Althougth the form.isValid should enter in the scope for the card...button should be disabled until the fields are valid.

Thoughts? Im gonna leave it for the moment as its easy for testing, but is the form validation on the card scope?